File: apk/app/src/main/java/com/safebrowser/app/DeviceAdminReceiver.java
package com.safebrowser.app;

import android.content.Context;
import android.content.Intent;
import android.util.Log;

/**
 * Diperlukan agar aplikasi dapat disetel sebagai device owner lewat ADB atau MDM,
 * yang memungkinkan lock task tanpa dialog konfirmasi peserta:
 *
 *   adb shell dpm set-device-owner com.safebrowser.app/.DeviceAdminReceiver
 */
public class DeviceAdminReceiver extends android.app.admin.DeviceAdminReceiver {

    private static final String TAG = "SafeBrowserAdmin";

    @Override
    public void onEnabled(Context context, Intent intent) {
        Log.i(TAG, "Device admin diaktifkan");
    }

    @Override
    public void onDisabled(Context context, Intent intent) {
        Log.i(TAG, "Device admin dinonaktifkan");
    }
}

File: apk/app/src/main/java/pochi/exam/clock/DeviceAdminReceiver.kt
package pochi.exam.clock

import android.app.admin.DeviceAdminReceiver
import android.content.Context
import android.content.Intent

class ExamClockDeviceAdmin : DeviceAdminReceiver() {
    override fun onEnabled(context: Context, intent: Intent) {}
    override fun onDisabled(context: Context, intent: Intent) {}
}
